### PR TITLE
Documentation changes

### PR DESCRIPTION
**Pull request by percygautam**
## Description
Modified the `extend_xr_method` to include docstring and examples. In continuation with the #1223 and #1254 and #1300. Related to #1066 .

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by percygautam** [comment]
I have added examples for custom extended methods and will start adding examples for extended methods. It will be helpful if it can get reviewed once as I continue adding examples.

File: arviz/data/base.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def make_attrs(attrs=None, library=None):
     return default_attrs
 
 
-def _extend_xr_method(func):
+def _extend_xr_method(func, doc=None, description=None, examples=None, see_also=None):
```
Comment (OriolAbril) [inline arviz/data/base.py:240]: strings are not mutable, so there is no need to do the `None` trick like with dicts or lists. You can directy set the default to `doc="",`...

File: arviz/data/base.py [hunk 2]
```diff
@@ -237,7 +237,7 @@ def make_attrs(attrs=None, library=None):
...
     """Make wrapper to extend methods from xr.Dataset to InferenceData Class."""
```
Comment (OriolAbril) [inline arviz/data/base.py:241]: can you document the different args here now that there are several?

File: arviz/data/base.py [hunk 3]
```diff
@@ -256,11 +256,14 @@ def wrapped(self, *args, **kwargs):
 
         return None if _inplace else out
 
-    description = """
-    This method is extended from xarray.Dataset methods. For more info see :meth:`xarray:xarray.Dataset.{method_name}`
+    description = "" if description is None else "\n" + description
+    description_default = """
+    {method_name} method is extended from xarray.Dataset methods.{description}
+    For more info see :meth:`xarray:xarray.Dataset.{method_name}`
```
Comment (OriolAbril) [inline arviz/data/base.py:262]: check the breaklines. with numpydoc, docstrings should have a single sentence in the first line (generally goes right after the """ too, not below), then one empty line and then the extended description. Same thing for breaklines between sections.

File: arviz/data/inference_data.py [hunk 4]
```diff
@@ -417,6 +448,39 @@ def stack(
             A new InferenceData object by default.
             When `inplace==True` perform selection in-place and return `None`
 
+        Examples
+        --------
+        Use ``stack`` to stack any number of existing dimensions into a single new dimension.. We first check the
```
Comment (OriolAbril) [inline arviz/data/inference_data.py:453]: Using some example to stack chain and draw dims into `sample` one will probably be easier to follow and understand for users

File: arviz/data/inference_data.py [hunk 5]
```diff
@@ -417,6 +448,39 @@ def stack(
...
+                    "b": np.random.randn(1, 100, 10),
+                    "c": np.random.randn(1, 100, 3, 4),
+                    }
+               ...: coords = {
+                        "c1": np.arange(3),
+                        "c99": np.arange(4),
+                        "b1": np.arange(10),
+                    }
+               ...: dims = {"c": ["c1", "c99"], "b": ["b1"]}
+               ...: dataset = from_dict(posterior=datadict, posterior_predictive=datadict, coords=coords, dims=dims)
```
Comment (OriolAbril) [inline arviz/data/inference_data.py:470]: you are importing arviz as az, thus this should be `az.from_dict`, I also recommend not using dataset due to confusion with xarray datasets, better use `data`, `idata`, the name of the model like radon or eight_schools...

**Pull request merged**